pkg/util/kube/pvc_pv.go

```diff
@@ -282,18 +282,20 @@ func WaitPVCConsumed(ctx context.Context, pvcGetter corev1client.CoreV1Interface
 			return false, errors.Wrapf(err, "error to get pvc %s/%s", namespace, pvc)
 		}
 
-		if tmpPVC.Spec.StorageClassName != nil && storageClass == nil {
-			storageClass, err = storageClient.StorageClasses().Get(ctx, *tmpPVC.Spec.StorageClassName, metav1.GetOptions{})
-			if err != nil {
-				return false, errors.Wrapf(err, "error to get storage class %s", *tmpPVC.Spec.StorageClassName)
+		if !ignoreConsume {
```

Comment:
I have a trivial comment, if ignoreConsume == true it seems we don't need to enter the `Pollxx` function, and the code will be slightly easier to read:
```
	tmpPVC, err := pvcGetter.PersistentVolumeClaims(namespace).Get(ctx, pvc, metav1.GetOptions{})
	if err != nil {
		return false, errors.Wrapf(err, "error to get pvc %s/%s", namespace, pvc)
	}
	if ignoreConsume {
                return "", tmpPVC, nil
        }
	err := wait.PollUntilContextTimeout(ctx, waitInternal, timeout, true, func(ctx context.Context) (bool, error) {
	        	if tmpPVC.Spec.StorageClassName != nil && storageClass == nil {
	        	.....
	        	updated = tmpPVC
	        	return true, nil
	        })
	        return selectedNode, updated, err
```

Reply:
There is a side benefit with the current code --- we could probe if the targetPVC exists by always calling `pvcGetter.PersistentVolumeClaims(namespace).Get`. If the targetPVC doesn't exists, we could fail the restore earlier.
For sure, to achieve the same purpose, we could realign the code for better understanding --- adding another `pvcGetter.PersistentVolumeClaims(namespace).Get` outside of `Pollxx`, but that will add redundant code.

So let's keep it as is for now.